Sources/Bot/Services/MergeService.swift

```diff
@@ -483,6 +498,20 @@ extension MergeService {
 
 extension MergeService {
 
+    public struct Trace: Equatable {
+        let function: String
+        let line: UInt
+
+        static func capture(function: String = #function, line: UInt = #line) -> Trace {
+            return self.init(function: function, line: line)
+        }
+
+        public static func ==(lhs: Trace, rhs: Trace) -> Bool {
+            // we don't really want to compare this in tests
+            return true
+        }
```

Comment:
Then why does it need to be equatable in the first place?

Reply:
for automatic code generation for conformance of enum cases?

Reply:
mmmh. Not sure it's worth it, since it could then lead to hard to understand and debug situations in the future in case we end up actually adding tests that do an `==` on two `Trace` objects for some reason…

Reply:
We shouldn't write such tests because they will be breaking on each refactoring with no value in that, and this is purely info for debugging.

Reply:
Well sure but with your shortcut, they won't break which they should 😜 Just feels quite weird to have an `Equatable` implementation that doesn't tell the truth is all 😉 🤷‍♂

Reply:
It feels weird to write such tests and to implement equality on whole State just because of this type